src/GraphQL.Tests/Caching/MemoryCacheTests.cs

```diff
@@ -194,4 +197,71 @@ public async Task ExecuteAsync(bool querySet, bool documentIdSet, bool docSet, b
         memoryDocumentCacheMock.Protected().Verify("GetAsync", getCalled ? Times.Once() : Times.Never(), ItExpr.IsAny<ExecutionOptions>());
         memoryDocumentCacheMock.Protected().Verify("SetAsync", setCalled ? Times.Once() : Times.Never(), ItExpr.IsAny<ExecutionOptions>(), ItExpr.IsAny<GraphQLDocument>());
     }
+
+    [Theory]
+    [InlineData("schema1", "query1", "schema1", "query1", true)]
+    [InlineData("schema1", "query1", "schema2", "query1", false)]
+    [InlineData("schema2", "query2", "schema2", "query2", true)]
+    [InlineData("schema2", "query2", "schema1", "query2", false)]
+    [InlineData("schema1", "query1", "schema2", "query2", false)]
```

Comment:
Ensures cache is unique across different `Schema` instances, where `schema.GetType() == typeof(Schema)`.